pkg/deploy/generator/scripts/gatewayVMSS.sh

```diff
@@ -150,7 +150,7 @@ touch /etc/containers/nodocker
 mkdir -p /root/.docker
 REGISTRY_AUTH_FILE=/root/.docker/config.json az acr login --name "$(sed -e 's|.*/||' <<<"$ACRRESOURCEID")"
 
-MDMIMAGE="${RPIMAGE%%/*}/${MDMIMAGE##*/}"
+MDMIMAGE="${RPIMAGE%%/*}/${MDMIMAGE#*/}"
```

Comment:
It's just nit, but I'm wondering if we can set it directly without substring removal. (maybe we need more code changes)
It was a bit difficult to imagine what `MDMIMAGE` would be like.

Reply:
This was difficult to understand for me as well, I agree that we potentially should do something different here. My understanding of the intent here is as follows:
- We set the `MDMIMAGE` env var on the VMSS to a value without a specified registry, e.g. `/genevamdm:${TAG}` before this change, and `/distroless/genevamdm:${TAG}` after. We don't know what image registry we're pulling from, as it's different depending on the environment the RP is running in. 
- In the code here, we extract the registry from the `RPIMAGE` env var, and prefix it to the MDM image name from above. The code before would only keep everything in the `MDMIMAGE` value "after the last `/`", my change here is to keep everything "after the first `/`" (to preserve the `/distroless/` prefix). 

I'll bring this up with @s-fairchild, as he is working on #3534 right now as a more comprehensive restructure of these VMSS scripts.

Reply:
Thank you for clarifying!
good to know that there is an ongoing refactor task.